### PR TITLE
chore(ci): Add CodeQL configuration

### PR DESCRIPTION
**Pull request by rhamzeh**
## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by jimmyjames — DISMISSED** [review]

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners (GitHub.com only)
+    # Consider using larger runners or machines with greater resources for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
```
Comment (ewanharris) [inline .github/workflows/codeql.yml:30]: No need for the swift check
```suggestion
    runs-on: ubuntu-latest'
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      # required to fetch internal or private CodeQL packs
+      packages: read
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (ewanharris) [inline .github/workflows/codeql.yml:44]: Do we need a matrix here given that we're just running for one language?

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    # If the analyze step fails for one of the languages you are analyzing with
+    # "We were unable to automatically build your code", modify the matrix above
+    # to set the build mode to "manual" for that language. Then modify this step
+    # to build your code.
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+    - if: matrix.build-mode == 'manual'
+      shell: bash
+      run: |
+        dotnet build --no-restore
```
Comment (ewanharris) [inline .github/workflows/codeql.yml:82]: Restore before build? (or maybe just drop the `--no-restore`?)
```suggestion
        dotnet restore
        dotnet build --no-restore
```

**Review by ewanharris — CHANGES_REQUESTED** [review]
Could we maybe remove some of the comments if they're not relevant?

**Review by ewanharris — APPROVED** [review]

**Pull request merged**